junit-vintage-engine/src/main/java/org/junit/vintage/engine/discovery/JUnit4DiscoveryRequestResolver.java

```diff
@@ -36,18 +36,16 @@
 public class JUnit4DiscoveryRequestResolver {
 
 	private static final IsPotentialJUnit4TestClass isPotentialJUnit4TestClass = new IsPotentialJUnit4TestClass();
-	private final EngineDescriptor engineDescriptor;
 	private final Logger logger;
 
-	public JUnit4DiscoveryRequestResolver(EngineDescriptor engineDescriptor, Logger logger) {
-		this.engineDescriptor = engineDescriptor;
+	public JUnit4DiscoveryRequestResolver(Logger logger) {
 		this.logger = logger;
 	}
 
-	public void resolve(EngineDiscoveryRequest discoveryRequest) {
+	public void resolveSelectors(EngineDiscoveryRequest discoveryRequest, TestDescriptor engineDescriptor) {
```

Comment:
It does not only resolve selectors but also applies filters. I think it does indeed resolve a complete `EngineDiscoveryRequest`.